api/src/main/java/org/xnio/ssl/JsseSslConduitEngine.java

```diff
@@ -317,7 +317,7 @@ private int wrap(final ByteBuffer src, boolean isCloseExpected) throws IOExcepti
                     doFlush();
                 }
             } catch (IOException ignore) {}
-            throw e;
+            throw new SSLException("Missing required intermediate / root certificates in a truststore: ", e);
```

Comment:
@moulalis Here an `IOException` is being wrapped with an `SSLException` with a specific message but we don't actually know if ""Missing required intermediate / root certificates in a truststore: " is an appropriate message for the exception that's occurred here. You could try checking with Brad Maxwell (since the issue was originally reported by him) to see specifically which exception should be caught and wrapped and under what circumstances if that's not already mentioned in the issue description. Thanks!

Reply:
> @moulalis Here an `IOException` is being wrapped with an `SSLException` with a specific message but we don't actually know if "Missing required intermediate / root certificates in a truststore: " is an appropriate message for the exception that's occurred here. You could try checking with Brad Maxwell (since the issue was originally reported by him) to see specifically which exception should be caught and wrapped and under what circumstances if that's not already mentioned in the issue description. Thanks!

Thank you @fjuma for the information.

Reply:
It looks like https://docs.oracle.com/javase/8/docs/api/java/security/cert/CertPathBuilderException.html is java API exception that we would be looking to catch some where, and then throw an exception that wraps the original with a message similar to what you have "unable to find valid certification path to requested target , check for missing required intermediate / root certificates in a truststore"

Reply:
Looking into it.